### PR TITLE
Fix two issues (typos) related to running JAX code with multiple GPUs

### PR DESCRIPTION
**Pull request by connection-on-fiber-bundles**
See discussion in https://github.com/deepmind/ferminet/issues/16

### CODE REVIEW

**Comment by connection-on-fiber-bundles** [comment]
@googlebot I signed it!

File: ferminet/train.py [hunk 1]
```diff
@@ -290,11 +290,8 @@ def train(cfg: ml_collections.ConfigDict):
 
   # Initialisation done. We now want to have different PRNG streams on each
   # device. Shard the key over devices
-  key, subkeys = jax.random.split(key, num_devices+1)
+  key, *subkeys = jax.random.split(key, num_devices+1)
   subkeys = jnp.stack(subkeys)
-  # Handle num_devices=1 case by explicitly broadcasting from an array of shape
-  # (2,) to an array of (num_devices, 2). This is a no-op for num_devices > 1.
-  subkeys = jnp.broadcast_to(subkeys, (num_devices, 2))
```
Comment (connection-on-fiber-bundles) [inline ferminet/train.py:297]: Seems not necessary for `num_devices == 1` any more because `subkeys` is also a list of jax array in that case now

**Review by jsspencer — APPROVED** [review]
Looks good, thanks!

**Comment by jsspencer** [comment]
Sorry this took a few days to merge. I just had to sort a few other things out first to make the workflow a bit more manageable.

**Pull request merged**